src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -213,6 +219,14 @@ def set_opts(self, curl, url, params, headers,
         else:
             raise Exception('Unsupported HTTP method "%s"' % verb)
 
+        try:
+            token = self.tmgr.getToken()
```

Comment:
It really depends how we decide to handle token creation. It is not the same as X509. Current procedure is to have separate process/daemon which will handle token creation. it is more like ssh-agent rather the voms-proxy-init. In former case, you deploy separate token agent which handle the tokens. In latter, you really obtain a new proxy.

Reply:
Yes, what I am saying is that if the system is defined to use tokens, it should use it instead of the x509 certificates, even if that token is expired!

So, I would basically take this code out of the try/except block.

Reply:
Resolving this comment as you have already implemented the suggested changes, thanks.